client/src/mid.rs

```diff
@@ -321,19 +334,73 @@ impl State {
 
 /// Init middleware state
 /// This function is called by UI to create the Middleware state and establish a connection to the Database.
-pub async fn init(url: &str) -> Result<State, reqwest::Error> {
-    let state = State {
+/// Important: Make sure `url` does not contain a trailing `/`
+#[tracing::instrument]
+pub async fn init(url: &str) -> color_eyre::Result<State> {
```

Comment:
I want to be able to assume that a `State` can only be created if there is a valid server connection. While there is a lot of code manually initializing the state now, this will change in the future when lazy loading is implemented.